perceval/components/processor.py

```diff
@@ -294,3 +302,28 @@ def _expand_TD_processor(components: list, backend_name: str, depth: int, m: int
         p.add(r, c)
     p.mode_post_selection(mode_post_select)
     return p
+
+
+def _find_equivalent(state):
+    if not state.has_annotations:
+        return state
+    annot_numbers = dict()
+    i = 0
+    new_state = "|"
+
+    for mode in range(state.m):
+        if not state[mode]:
+            new_state += "0,"
+            continue
+        annotations = state.get_mode_annotations(mode)
+        for n in range(state[mode]):
+            annot = annotations[n]["_"]
```

Comment:
This line searches for "_" annotation types and makes this method specific to this kind of annotation. I understand it works on source-generated StateVectors.
However, is there a way to make `_find_equivalent` more generic?